v2/spiffetls/tlsconfig/config.go

```diff
@@ -8,63 +8,122 @@ import (
 	"github.com/spiffe/go-spiffe/v2/svid/x509svid"
 )
 
+type verifyPeerCertificate = func(raw [][]byte, certs [][]*x509.Certificate) error
+
+func wrapVerifyPeerCertificate(wrapped, newVerifier verifyPeerCertificate) verifyPeerCertificate {
+	if wrapped == nil {
+		return newVerifier
+	}
+
+	return func(raw [][]byte, certs [][]*x509.Certificate) error {
+		if err := newVerifier(raw, certs); err != nil {
+			return err
+		}
+		return wrapped(raw, certs)
+	}
+}
+
 // TLSClientConfig returns a TLS configuration which verifies and authorizes
 // the server X509-SVID.
 func TLSClientConfig(svid x509svid.Source, bundle x509bundle.Source, authorizer Authorizer) *tls.Config {
-	panic("not implemented")
+	config := new(tls.Config)
+	HookTLSClientConfig(config, svid, bundle, authorizer)
+	return config
 }
 
 // HookTLSClientConfig sets up the TLS configuration to verify and authorize
 // the server X509-SVID. If there is an existing callback set for
 // VerifyPeerCertificate it will be wrapped by by this package and invoked
 // after SPIFFE authentication has completed.
 func HookTLSClientConfig(config *tls.Config, svid x509svid.Source, bundle x509bundle.Source, authorizer Authorizer) {
-	panic("not implemented")
+	config.InsecureSkipVerify = true
+	config.GetCertificate = GetCertificate(svid)
+	config.VerifyPeerCertificate = wrapVerifyPeerCertificate(config.VerifyPeerCertificate, VerifyPeerCertificate(bundle, authorizer))
+
+	// Not used by clients
+	config.ClientAuth = tls.NoClientCert
+	config.GetClientCertificate = nil
+	config.Certificates = nil
+	config.RootCAs = nil
+	config.NameToCertificate = nil
 }
 
 // MTLSClientConfig returns a TLS configuration which presents an X509-SVID
 // to the server and verifies and authorizes the server X509-SVID.
 func MTLSClientConfig(svid x509svid.Source, bundle x509bundle.Source, authorizer Authorizer) *tls.Config {
-	panic("not implemented")
+	config := new(tls.Config)
+	HookMTLSClientConfig(config, svid, bundle, authorizer)
+	return config
 }
 
 // HookMTLSClientConfig sets up the TLS configuration to present an X509-SVID
 // to the server and verify and authorize the server X509-SVID. If there is an
 // existing callback set for VerifyPeerCertificate it will be wrapped by by
 // this package and invoked after SPIFFE authentication has completed.
 func HookMTLSClientConfig(config *tls.Config, svid x509svid.Source, bundle x509bundle.Source, authorizer Authorizer) {
-	panic("not implemented")
+	config.InsecureSkipVerify = true
+	config.GetCertificate = GetCertificate(svid)
```

Comment:
Clients use `GetClientCertificate` to provide the certificate... this should be swapped with `GetClientCertificate` below...